esp-hal/src/sha.rs

```diff
@@ -120,8 +120,8 @@ impl crate::interrupt::InterruptConfigurable for Sha<'_> {
 }
 
 // A few notes on this implementation with regards to 'memcpy',
-// - It seems that ptr::write_bytes already acts as volatile, while ptr::copy_*
-//   does not (in this case)
+// - It seems that ptr::write already acts as volatile, while ptr::copy_* does
```

Comment:
I'd probably just remove this line completely. `ptr::write` is not volatile access, regardless of how it acts.